api/gp_lttb.md

```diff
@@ -0,0 +1,95 @@
+---
+api_name: gp_lttb()
+excerpt: An implementation of our lttb algorithm that will preserve gaps in the original data
+topics: [hyperfunctions]
+keywords: [downsample, smooth, hyperfunctions, toolkit]
+api:
+  license: community
+  type: function
+  experimental: true
+  toolkit: true
+hyperfunction:
+  family: downsample
+  type: one-step aggregate
+# fields below will be deprecated
+api_category: hyperfunction
+api_experimental: true
+toolkit: true
+hyperfunction_family: 'downsample'
+hyperfunction_subfamily: 'downsample'
+hyperfunction_type: other
+---
+
+# gp_lttb()  <tag type="toolkit">Toolkit</tag><tag type="experimental-toolkit">Experimental</tag>
+
+Gap preserving LTTB is a specialization of the [LTTB][lttb] algorithm that
+preserves gaps in the underlying data. It uses a minimum gap
+size, which can be provided by the caller, or that defaults to the time span of the
+incoming data divided by the desired target resolution. It then scans through
+the input and breaks it into ranges of points separated by gaps of at least the
+minimum size. It then performs a LTTB on each range, with resolution
+proportional to the percentage of the total points with that range, and
+concatenates the results.
+
+If the minimum gap size is set too low, the algorithm might produce more points
+than the target resolution. This occurs if there are more ranges than half the
+target resolution, because each range returns at least two points.
+
+## Required arguments
+
+|Name|Type|Description|
+|-|-|-|
+|`time`|`TIMESTAMPTZ`|Time (x) value for the data point|
+|`value`|`DOUBLE PRECISION`|Data (y) value for the data point|
+|`gapsize`|`INTERVAL`|(Optional) Minimum gap size to divide input on|
```

Comment:
Could you move this to an "Optional arguments" table instead? It's a good question whether we actually need two separate tables but let's keep it consistent across all the API references for now

Reply:
Updated